File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/atcoder_ABC104_C.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int D = scanner.nextInt();
        int G = scanner.nextInt();
        List<int[]> PC = new ArrayList<>();
        for (int i = 0; i < D; i++) {
            int p = scanner.nextInt();
            int c = scanner.nextInt();
            PC.add(new int[]{p, c});
        }
        scanner.close();

        int ans = 1000000000;
        for (int i = 0; i < (1 << D); i++) {
            int score = 0;
            int problem = 0;
            for (int j = 0; j < D; j++) {
                if (((i >> j) & 1) == 1) {
                    score += 100 * (j + 1) * PC.get(j)[0] + PC.get(j)[1];
                    problem += PC.get(j)[0];
                }
            }
            if (score > G) {
                continue;
            }
            int left = G - score;
            for (int j = 0; j < D; j++) {
                if (((i >> j) & 1) == 1) {
                    continue;
                }
                if (left > 100 * (j + 1) * PC.get(j)[0] + PC.get(j)[1]) {
                    continue;
                } else {
                    int tmp = (left + (100 * (j + 1)) - 1) / (100 * (j + 1));
                    ans = Math.min(ans, problem + Math.min(tmp, PC.get(j)[0]));
                }
            }
        }
        System.out.println(ans);
    }
}

File: dataset/avatar/Python/Code/atcoder_ABC104_C.py
D , G = map ( int , input ( ).split ( ) )
PC = [ tuple ( map ( int , input ( ).split ( ) ) ) for _ in range ( D ) ]
ans = 10 ** 9
for i in range ( 2 ** D ) :
    score = 0
    problem = 0
    for j in range ( D ) :
        if ( i >> j ) & 1 :
            score += 100 * ( j + 1 ) * PC [ j ] [ 0 ] + PC [ j ] [ 1 ]
            problem += PC [ j ] [ 0 ]
    if score > G :
        continue
    left = G - score
    for j in range ( D ) :
        if ( i >> j ) & 1 :
            continue
        if left > 100 * ( j + 1 ) * PC [ j ] [ 0 ] + PC [ j ] [ 1 ] :
            pass
        else :
            tmp = ( left + ( 100 * ( j + 1 ) ) - 1 ) // ( 100 * ( j + 1 ) )
            ans = min ( ans , problem + min ( tmp , PC [ j ] [ 0 ] ) )
print ( ans )

